packages/react-components/react-toast/src/state/vanilla/toaster.ts

```diff
@@ -0,0 +1,80 @@
+import { Toast, ToastEventMap, ToastId, ToastOptions } from '../types';
+import { EVENTS } from '../constants';
+
+// TODO convert to closure
+export class Toaster {
+  public visibleToasts: Set<ToastId>;
+  public toasts: Map<ToastId, Toast>;
+  public onUpdate: () => void;
+  private targetDocument: Document;
+
+  private listeners = new Map<keyof ToastEventMap, Function>();
+
+  constructor(targetDocument: Document) {
+    this.toasts = new Map<ToastId, Toast>();
+    this.visibleToasts = new Set<ToastId>();
+    this.onUpdate = () => null;
+    this.targetDocument = targetDocument;
+
+    this._initEvents();
+  }
+
+  public dispose() {
+    for (const [event, callback] of this.listeners.entries()) {
+      this._removeEventListener(event, callback as () => void);
+    }
+  }
+
+  public isToastVisible(toastId: ToastId) {
+    return this.visibleToasts.has(toastId);
+  }
+
+  private _initEvents() {
+    const buildToast: (e: ToastEventMap[typeof EVENTS.show]) => void = e => this._buildToast(e.detail);
```

Comment:
```suggestion
    const buildToast = (e: ToastEventMap[typeof EVENTS.show]) => this._buildToast(e.detail);
```

nit: There is no sense to define type for this function